packages/css-framework/src/components/_data-list.scss

```diff
@@ -0,0 +1,119 @@
+@use "../abstracts/functions" as f;
+@use "../config" as config;
+
+.rn-data-list {
+  &.is-collapsible {
+    border-radius: 3px;
+
+    .rn-data-list {
+      &__header {
+        padding: f.spacing("3") f.spacing("2") f.spacing("3") f.spacing("2");
+        background-color: f.color("neutral", "white");
+        border-bottom: 1px solid f.color("neutral", "100");
+        transition: color config.$animation-timing;
+        cursor: pointer;
+        outline: none;
+        pointer-events: all;
+        &:hover {
+          background-color: f.color("neutral", "000");
+          .rn-data-list__action {
+            background-color: f.color("neutral", "200");
+          }
+        }
+      }
+
+      &__badge {
+        display: inline-block;
+        margin-left: f.spacing("4");
+      }
+
+      &__action {
+        border-radius: 3px;
+        padding: f.spacing("2") f.spacing("2") f.spacing("2") f.spacing("2");
+        display: flex;
+        background: f.color("neutral", "100");
+        transition: background-color config.$animation-timing;
+        svg {
+          transform: translate(-1px, 0);
+          transition: transform config.$animation-timing;
+        }
+      }
+
+      &__sheet {
+        transform: scale(1, 0);
+        transform-origin: 0 0;
+        height: 0;
+        overflow: hidden;
+        transition: all config.$animation-timing;
+        background-color: f.color("neutral", "white");
+      }
+      &__item {
+        padding: f.spacing("4") f.spacing("4");
+      }
+    }
+
+    &.is-expanded {
+      .rn-data-list {
+        &__action {
+          svg {
+            transform: rotate(180deg) translate(0, 1px);
+          }
+        }
+        &__sheet {
+          transform: scale(1, 1);
+          height: unset;
+          overflow: hidden;
+        }
+      }
+    }
+  }
+
+  &__sheet {
```

Comment:
Can we write the full selector instead of doing the & trick? 

It can be difficult to search for selectors in a codebase.

Reply:
It's fine for pseudo selectors though.

```css
&:hover { ... } 
```